crates/nu-command/src/filesystem/cp.rs

```diff
@@ -355,8 +355,18 @@ fn copy_file(src: PathBuf, dst: PathBuf, span: Span) -> Value {
             let message = format!("copy file {src:?} failed: {e}");
 
             use std::io::ErrorKind;
+            let dst_info = std::fs::metadata(&dst);
```

Comment:
I don't think this does the right thing. This will return `Ok` if and only if the file or directory exists and the path leading there is readable for the user. There are several failure modes that are not covered by this: e.g. permissions and that the target doesn't exist is reasonable for `cp` without flags.